### DISCUSSION TITLE
Explanation of why the pdf.min.js file comes to my browser when I open pdf using react-pdf

### DISCUSSION THREAD

**Discussion by ashwinj984** [wojtekmaj/react-pdf#1561] [Q&A] [answered]
One question @wojtekmaj,
We do not send our node_modules to the github, server makes one for us at the time of deployment after reading the dependencies and transitive dependencies from the package.json and yarn.lock. So when we open our web-app in our browser then how the pdf.js file comes to our browser as I can see it in my sources tab in browser.

Can you give me a explanation for it? How a node_modules file is getting opened up in the browser because our server has the node_modules not our browser.

<img width="1463" alt="Screenshot 2023-07-13 at 11 22 58 PM 1" src="https://github.com/wojtekmaj/react-pdf/assets/56488757/0f3cebf9-89d7-45d8-8dae-a4615a794deb">

**Comment by wojtekmaj** [comment]
Can't say without seeing your setup in detail. 

pdf.js file is a dependency module like any other, so it will be included in your bundle. Depending on your configuration, it may or may not be packed with other modules your app depends on. What you're looking at may also be a result of source maps work, showing you actual source files without them actually being there in unchanged form.

pdf.worker.js is the one that's special - worker file needs to be a separate, static file, untouched by bundlers.

**Comment by ashwinj984** [comment]
Sorry I actually added  the localhost pic last time, this is the production image as you can see when I am opening the pdf a file called pdf.worker.min.54da87bc.js coming in my threads section why is that?
<img width="1466" alt="Screenshot 2023-07-15 at 8 55 36 PM" src="https://github.com/wojtekmaj/react-pdf/assets/56488757/602721cf-6a35-4f0b-98f0-a86449aa03fb">

**Reply by wojtekmaj** [reply, marked as answer]
Well because you're supposed to. PDFs are rendered off the main thread if possible. All good.

**Reply by ashwinj984** [reply]
So basically you are saying that when we open a pdf, our browser sends a request to the server to fetch this pdf worker file and that's how it gets rendered off the main thread?

**Reply by wojtekmaj** [reply]
Yes.
